File: apk/app/src/main/java/com/example/momentra/ui/setup/SetupDropdownField.kt
package com.example.momentra.ui.setup

import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.DropdownMenu
import androidx.compose.material3.DropdownMenuItem
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.remember
import androidx.compose.runtime.setValue
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.platform.testTag
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.example.momentra.ui.shell.maestro.MaestroIds

@Composable
fun SetupDropdownField(
    label: String,
    hint: String? = null,
    value: String,
    options: List<String>,
    onValueChange: (String) -> Unit,
    modifier: Modifier = Modifier,
    accent: Color = SetupTokens.BrandPrimary,
    borderColor: Color = Color(0xFF1E293B),
    cardColor: Color = SetupTokens.BizCard,
    testTag: String = MaestroIds.SETUP_DROPDOWN,
) {
    var expanded by remember { mutableStateOf(false) }

    Column(
        modifier = modifier
            .fillMaxWidth()
            .padding(vertical = 6.dp)
            .testTag(testTag),
        verticalArrangement = Arrangement.spacedBy(8.dp),
    ) {
        Column(verticalArrangement = Arrangement.spacedBy(4.dp)) {
            Text(label, color = SetupTokens.TextPrimary, fontWeight = FontWeight.SemiBold, fontSize = 14.sp)
            if (hint != null) {
                Text(hint, color = SetupTokens.TextSecondary, fontSize = 12.sp)
            }
        }
        Box(modifier = Modifier.fillMaxWidth()) {
            Row(
                modifier = Modifier
                    .fillMaxWidth()
                    .clip(RoundedCornerShape(12.dp))
                    .background(cardColor)
                    .border(1.dp, if (expanded) accent else borderColor, RoundedCornerShape(12.dp))
                    .clickable { expanded = true }
                    .padding(horizontal = 14.dp, vertical = 12.dp)
                    .testTag("$testTag.trigger"),
                horizontalArrangement = Arrangement.SpaceBetween,
                verticalAlignment = Alignment.CenterVertically,
            ) {
                Text(
                    text = value.ifBlank { "Select" },
                    color = if (value.isBlank()) SetupTokens.TextSecondary else Color.White,
                    fontSize = 15.sp,
                )
                Text("▼", color = SetupTokens.TextSecondary, fontSize = 12.sp)
            }
            DropdownMenu(
                expanded = expanded,
                onDismissRequest = { expanded = false },
                modifier = Modifier
                    .background(SetupTokens.BizCard)
                    .testTag("$testTag.menu"),
            ) {
                options.forEach { option ->
                    DropdownMenuItem(
                        text = {
                            Text(
                                option,
                                color = if (option == value) accent else Color.White,
                                fontWeight = if (option == value) FontWeight.Bold else FontWeight.Normal,
                            )
                        },
                        onClick = {
                            onValueChange(option)
                            expanded = false
                        },
                    )
                }
            }
        }
    }
}

File: momentra/momentra/Shell/Setup/SetupDropdownField.swift
import SwiftUI

struct SetupDropdownField: View {
    let label: String
    var hint: String? = nil
    let options: [String]
    @Binding var value: String
    var accent: Color = SetupTokens.brandPrimary
    var testTag: String = "setup.dropdown"

    var body: some View {
        VStack(alignment: .leading, spacing: 8) {
            VStack(alignment: .leading, spacing: 4) {
                Text(label)
                    .font(.system(size: 14, weight: .semibold))
                    .foregroundStyle(SetupTokens.textPrimary)
                if let hint {
                    Text(hint)
                        .font(.system(size: 12))
                        .foregroundStyle(SetupTokens.textSecondary)
                }
            }
            HStack {
                Spacer()
                Menu {
                    ForEach(options, id: \.self) { option in
                        Button(option) { value = option }
                    }
                } label: {
                    HStack(spacing: 8) {
                        Text(value.isEmpty ? "Select" : value)
                            .font(.system(size: 15, weight: .semibold))
                            .foregroundStyle(value.isEmpty ? SetupTokens.textSecondary : .white)
                        Text("▼")
                            .font(.system(size: 12))
                            .foregroundStyle(SetupTokens.textSecondary)
                    }
                    .frame(maxWidth: .infinity, alignment: .trailing)
                    .padding(.horizontal, 14)
                    .padding(.vertical, 12)
                    .background(SetupTokens.bizCard, in: RoundedRectangle(cornerRadius: 12, style: .continuous))
                    .overlay(
                        RoundedRectangle(cornerRadius: 12, style: .continuous)
                            .stroke(Color(hex: "#1E293B"), lineWidth: 1)
                    )
                }
                .accessibilityIdentifier("\(testTag).trigger")
            }
        }
        .padding(.vertical, 6)
        .accessibilityIdentifier(testTag)
    }
}
